sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriterMetricSuite.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources
+
+import java.util.Collections
+
+import org.scalatest.BeforeAndAfter
+import org.scalatest.time.SpanSugar._
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.connector.catalog.{Identifier, InMemoryTableCatalog}
+import org.apache.spark.sql.functions.lit
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.StructType
+
+class FileFormatDataWriterMetricSuite
```

Comment:
For this one, I guess we need @gengliangwang 's review since he requested?

Reply:
Yea, I think so.

Reply:
IIUC this one is for `InMemoryTable` instead of `FileFormatDataWriter`. Shall we rename the test suite?
As for testing `FileFormatDataWriter`, if it is tricky and difficult, let's mark it as follow up since the code changes in `FileFormatDataWriter.scala` look good.

Reply:
Ah, right, I was writing tests for custom `FileTable` which uses custom metrics but may copying some duplicated codes for `FileWrite`, etc. So in the end I turned to `InMemoryTable` but forgot that `InMemoryTable` doesn't use `FileWriterFactory`...

I will rename this test suite to `InMemoryTableMetricSuite`.

Let me do a follow up adding tests for `FileTable`. Thanks.